rasterio/_warp.pyx

```diff
@@ -582,7 +582,8 @@ def _reproject(
     log.debug("Setting NUM_THREADS option: %d", num_threads)
 
     if init_dest_nodata:
-        warp_extras = CSLSetNameValue(warp_extras, "INIT_DEST", "NO_DATA")
+        warp_extras = CSLSetNameValue(warp_extras, "INIT_DEST",
+                                      "NO_DATA" if dst_nodata is not None else "0")
```

Comment:
Should the default be `NaN` for floating point bands, or is the intention to initialize with something that looks like a legal value?

Reply:
I agree, getting a new Rasterio release out is more important than refining these things.